src/Components/WebAssembly/WebAssembly/src/Services/LazyAssemblyLoader.cs

```diff
@@ -53,39 +51,47 @@ public async Task<IEnumerable<Assembly>> LoadAssembliesAsync(IEnumerable<string>
             return await LoadAssembliesInServerAsync(assembliesToLoad);
         }
 
-        private Task<IEnumerable<Assembly>> LoadAssembliesInServerAsync(IEnumerable<string> assembliesToLoad)
+        private async Task<IEnumerable<Assembly>> LoadAssembliesInServerAsync(IEnumerable<string> assembliesToLoad)
         {
-            var loadedAssemblies = _loadedAssemblyCache.Where(assembly =>
-                assembliesToLoad.Contains(assembly.GetName().Name + ".dll"));
+            var loadedAssemblies = new List<Assembly>();
 
-            if (loadedAssemblies.Count() != assembliesToLoad.Count())
+            try
+            {
+                foreach (var assemblyName in assembliesToLoad)
+                {
+                    loadedAssemblies.Add(Assembly.Load(Path.GetFileNameWithoutExtension(assemblyName)));
+                }
+            }
+            catch (FileNotFoundException ex)
             {
-                var unloadedAssemblies = assembliesToLoad.Except(loadedAssemblies.Select(a => a.GetName().Name + ".dll"));
-                throw new InvalidOperationException($"Unable to find the following assemblies: {string.Join(",", unloadedAssemblies)}. Make sure that the appplication is referencing the assemblies and that they are present in the output folder.");
+                throw new InvalidOperationException($"Unable to find the following assembly: {ex.FileName}. Make sure that the appplication is referencing the assemblies and that they are present in the output folder.");
             }
 
-            return Task.FromResult(loadedAssemblies);
+            return await Task.FromResult(loadedAssemblies);
```

Comment:
Why await this? The earlier version seemed fine

Reply:
There's some weirdness with how the implicit type conversion is done now that we aren't instantiating `loadedAssemblies` from a LINQ query ([ref](https://stackoverflow.com/questions/25182011/why-async-await-allows-for-implicit-conversion-from-a-list-to-ienumerable)).

Reply:
Specifying the generic parameter (`return Task.FromResult<IEnumerable<Assembly>>(loadedAssemblies)`) is another option if you don't want to make the method `async`.

Reply:
Cool! Thanks for pointing this out.